### DISCUSSION TITLE
Text2Cypher：大语言模型驱动的图谱查询生成 - siwei.io

### DISCUSSION THREAD

**Discussion by giscus** [siwei-io/siwei-io.github.io#28] [Comments]
# Text2Cypher：大语言模型驱动的图谱查询生成 - siwei.io

如何在 NebulaGraph 上对知识图谱的自然语言查询。

https://www.siwei.io/llm-text-to-nebulagraph-query/

**Comment by ZzzybEric** [comment]
大佬，请问有试过针对中文场景嘛，效果如何

**Reply by wey-gu** [reply]
会有文章哈，给你先看下 https://www.siwei.io/demo-dumps/local-llm/Graph_RAG_Local.html

**Reply by ZzzybEric** [reply]
感谢大佬！！

**Reply by ZzzybEric** [reply]
大佬还有一个问题，NebulaGraphQAChain.from_llm方法的第一个api参数我可以用自己本地部署的大模型吗

**Reply by wey-gu** [reply]
应该是可以的，但是也要针对不同的 llm 去覆盖 prompt 才行

**Comment by wey-gu** [comment]
> 大佬还有一个问题，NebulaGraphQAChain.from_llm方法的第一个api参数我可以用自己本地部署的大模型吗

理论上可以，不过 prompt 可能要 override 调试一下哈。

**Comment by aqzwy** [comment]
老师我先请教下，为什么 llama_index 和langchain两种框架下，nl2Cypher 对应的方法KnowledgeGraphQueryEngine 和NebulaGraphQAChain 对应的propmt所描述的模板为什么不是一样的，这其中会有什么隐含的影响吗，我在本地测试时会发现前者框架构建出的语句 更优于后者

**Reply by wey-gu** [reply]
抱歉 NebulaGraphQAChain 中的设计有几个明显的优化落后了，我准备抽时间去优化一下哈

**Reply by aqzwy** [reply]
好的 老师 那我明白了

**Reply by xcxinng** [reply]
“NebulaGraphQAChain 中的设计有几个明显的优化落后了，我准备抽时间去优化一下哈”

**Reply by xcxinng** [reply]
老师有计划准备优化NebulaGraphQAChain了吗

**Comment by ccp123456789** [comment]
text2cypher的时候， 如果schema很长怎么处理呢？ 项目里面也没有介绍策略，因为图谱的点和边，以及属性这些都是需要传入prompt，很长的schema甚至有数千数万token

**Comment by BBDooooog** [comment]
老师，你好。NebulaGraphQAChain 对NebulaGraph的版本有要求吗？哪里可以找到相关信息呢。

**Comment by xcxinng** [comment]
体验了下langchain方式，目前中文感觉效果不理想，英文效果好很多，对于没有体验过text2SQL的我来说已经够惊艳了。然后还有一点是NebulaGraphQAChain的prompt感觉还是得需要自己去调整，从个人体验来说还没有到能真正开箱即用的效果～ 期待后面继续迭代优化
